api/src/main/java/org/pmiops/workbench/db/dao/NotebookCohortCacheService.java

```diff
@@ -0,0 +1,18 @@
+package org.pmiops.workbench.db.dao;
+
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.Workspace;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+
+public interface NotebookCohortCacheService {
+
+  NotebookCohortCacheDao getDao();
+
+  void updateNotebook(Workspace workspace, User user, String notebookName, Timestamp lastAccessDateTime);
```

Comment:
nit: I would probably use slightly different language to clarify that this refers to the cache entry rather than the notebook itself:

Maybe:

```
putNotebook()
putCohort()
removeNotebook()
```

or if you like update/delete:

```
updateNotebookEntry()
updateCohortEntry()
deleteNotebookEntry()
```